src/BackgroundProcess/Adapter/Out/Wp/WpBackgroundJobQueue.php

```diff
@@ -3,166 +3,108 @@
 
 namespace Jetty\BackgroundProcessing\BackgroundProcess\Adapter\Out\Wp;
 
-/**
- * WP Background Process
- *
- * @package WP-Background-Processing
- */
+use Jetty\BackgroundProcessing\BackgroundProcess\Application\Port\Out\QueueBatchRepository;
+use Jetty\BackgroundProcessing\BackgroundProcess\Domain\BackgroundJobQueue;
 
 /**
- * Abstract WP_Background_Process class.
- *
- * @abstract
- *
- * @extends WP_Async_Request
+ * Defines a background job queue that operates on multiple pieces in the
+ * background.
  */
-abstract class WpBackgroundProcess extends WpAsyncRequest
+abstract class WpBackgroundJobQueue extends WpAjaxHandler implements BackgroundJobQueue
 {
     /**
-     * Action
-     *
-     * (default value: 'background_process')
-     *
      * @var string
-     *
-     * @access protected
-     */
-    protected $action = 'background_process';
-
-    /**
-     * Start time of current process.
-     *
-     * (default value: 0)
-     *
-     * @var int
-     *
-     * @access protected
-     */
-    protected $start_time = 0;
-
-    /**
-     * Cron_hook_identifier
-     *
-     * @var mixed
-     *
-     * @access protected
      */
-    protected $cron_hook_identifier;
+    private $identifier;
 
     /**
-     * Cron_interval_identifier
-     *
-     * @var mixed
-     *
-     * @access protected
+     * @var QueueBatchRepository
      */
-    protected $cron_interval_identifier;
+    private $batchRepository;
 
     /**
      * Initiate new background process
      */
-    public function __construct()
+    public function __construct(string $actionName)
     {
-        parent::__construct();
+        $this->identifier = $actionName;
+        parent::__construct($actionName);
 
         $this->cron_hook_identifier     = $this->identifier . '_cron';
         $this->cron_interval_identifier = $this->identifier . '_cron_interval';
 
-        add_action($this->cron_hook_identifier, [ $this, 'handle_cron_healthcheck' ]);
-        add_filter('cron_schedules', [ $this, 'schedule_cron_healthcheck' ]);
+        add_action($this->cron_hook_identifier, function(): void {
+            $this->handleCronHealthcheck();
+        });
+        add_filter('cron_schedules', function($schedules): void {
+            $this->scheduleCronHealthcheck($schedules);
+        });
+
+        $this->batchRepository = new WpBatchItemRepository($actionName);
     }
 
 
-    public function dispatch(): array
+    final public function dispatch(array $data = []): array
     {
         // Schedule the cron healthcheck.
-        $this->schedule_event();
+        $this->scheduleEvent();
 
         // Perform remote post.
-        return parent::dispatch();
+        $request = new WpAjaxRequest($this->identifier);
+        return $request->dispatch($data);
```

Comment:
To me, it doesn't make much sense to have two different methods to define data in this class. Either data should be saved to the database, or it should be passed via an Ajax request. Not both.

Also, from my point of view, `AsyncRequest->dispatch(array)` is sufficient for describing this behavior (whatever its implementation happens to be). This Asynchronous operation Request does not intrinsically have an Ajax request. Or is it supposed to be? And then, does it need to be? Because if I want to start an Async operation, and I "dispatch" it, providing it the data I intend for it to process in the background, why would there ever be an alternative way to provide the data? And, yes, I know that POST requests have a limit to the amount of data that can be passed there. So, why wouldn't this implementation just save the data to the DB?

Just my $0.02.